public/js/log.js

```diff
@@ -60,250 +59,295 @@ MAGI_log.resizes = {
 
-$('#magi-loggingReadMore').click(function(e) {
-  e.preventDefault();
-  var isVisible = 'visible' == $('#magi-loggingReadMoreDiv').css('visibility'),
-      visibleState = isVisible ? 'hidden' : 'visible',
-      displayState = isVisible ? 'none' : 'block';
-
-  $('#magi-loggingReadMoreDiv').css('visibility', visibleState);
-  $('#magi-loggingReadMoreDiv').css('display', displayState);
-
+$().ready(function () {
+  $.get('/logEnabled', function(res, status) {
+    if(status == "success" && res === true) {
+      loggingEnabled = true;
+      initLogging();
+    } else {
+      loggingEnabled = false;
+    }
+  });
 });
 
-$('#magiConsentViewToggle').change(function() {
-  var checked = $('#magiConsentViewToggle').is(':checked');
-  loggingEnabled = checked;
-});
 
-$(document).keydown(function(e) {
-  if(e.ctrlKey || e.metaKey) {
-    extendLogEvents();
-  }
-});
+function initLogging() {
+  lastSentTime = Date.now();
+  MAGI_log.sessionLogStart = Date.now();
+  startLog();
 
-$(document).mousemove(function(e) {
-  addToLog(e, 'm');
-  if(e.pageX <= 0 || e.pageY <= 0) {
-    extendLogEvents();
-  }
-});
+  // append consent pane
+  var consentPane = d3.select('#view').append('div').style({
+    'border-top': '1px solid #eee',
+    'margin-left': '20px',
+    'margin-right': '200px',
+    'margin-top': '100px',
+    'min-width': '400px',
+    'max-width': '600px',
+    'padding-top': '15px'
+  });
+  consentPane.append('input')
+      .attr('id', 'magiConsentViewToggle')
+      .attr('type', 'checkbox')
+      .style({ 'display': 'inline-block', 'margin-right': '5px' });
+  consentPane.append('p')
+      .style('color', 'rgb(200,200,200)')
+      .style('display', 'inline-block')
+      .html('I do not want to improve the development of MAGI by sending usage information. <a href="" id="magi-loggingReadMore" style="color:rgb(200,200,225)">(Read More)</a>');
+  var consentPaneDesc = consentPane.append('div').attr('id', 'magi-loggingReadMoreDiv').append('blockquote');
+  consentPaneDesc.append('p').text("We don't track identifying information and we anonymize queries. This means that our logs don't track your name or any other information about who you are, and the genes and datasets you look at are scrambled to respect research privacy. Logs are made up of anonymized query information and interaction information, like how visualizations are being used.".replace(/\n/g, ''));
+  consentPaneDesc.append('p').text("Why collect this information? Cancer researchers are hard to observe in-lab due to how spread out geographically they are, which makes developing usable visualization software challenging. We use information about how you use the tool to understand how the tool is being used, to identify usability bottlenecks, and come up with new features.".replace(/\n/g, ''));
+
+  $('#magi-loggingReadMore').click(function(e) {
+    e.preventDefault();
+    var isVisible = 'visible' == $('#magi-loggingReadMoreDiv').css('visibility'),
```

Comment:
This whole `<div>` looks like it's at half opacity to me. Is that intentional?
![screen shot 2015-05-11 at 9 33 29 pm](https://cloud.githubusercontent.com/assets/2043206/7578629/6b82f61c-f825-11e4-9333-fe2a16e8aea4.png)